### DISCUSSION TITLE
A design system for the Wagtail ecosystem

### DISCUSSION THREAD

**Discussion by tbrlpld** [wagtail/wagtail#8208] [Ideas]
I have worked a little bit on the [Wagtail Sphinx theme](https://github.com/wagtail/sphinx_wagtail_theme) and there is a nice [design](https://www.figma.com/proto/hFjWqQeEjGd8NtTsEHcc4i/Wagtail-Documentation-(wagtail.io)?node-id=1%3A2&viewport=983%2C953%2C1&scaling=min-zoom&hide-ui=1) for it. 

But, there are still a tonne of open styling related issues on the theme repo. And some things, like hover states etc. are not covered in the designs. 

Also, there are wagtail.org and madewithwagtail.org that both define their own styles. Some are similar to the docs theme, some totally different. 

I am wondering if it would make sense to create a design system for the Wagtail ecosystem that can help inform the design of these secondary projects. Or should these maybe share the design system with the Wagtail application? 

This could help build the brand of Wagtail, as @vsalvino mentioned in the Wagtail Space talk today. 

Wondering what the communities ideas are here. 

*** 

# Existing designs

Here are a few impressions of the existing designs in the ecosystem.

Wagtail 3.0 admin
![admin](https://user-images.githubusercontent.com/24797493/163416883-78b3ee45-1d25-430c-8eb5-1a031e2ffe01.png)

Wagtail docs
![docs](https://user-images.githubusercontent.com/24797493/163416961-a2b7eb45-e07a-4166-a8ab-e18a3dc3caa2.png)

Wagtail.org
![org](https://user-images.githubusercontent.com/24797493/163417150-f7a62925-def2-423b-b247-f4d012a8df9e.png)

This week in Wagtail
![twiw](https://user-images.githubusercontent.com/24797493/163417176-0da1da0a-7f9e-4cc5-9cb5-315efdc9e6cf.png)

Made with Wagtail
![mww](https://user-images.githubusercontent.com/24797493/163417212-d673ccf0-8d5f-45fa-b34e-689d4bbce737.png)

Wagtail Localize docs
![localize](https://user-images.githubusercontent.com/24797493/163417306-8488694f-f4d3-446e-93ef-3b33afbb19f4.png)

Wagtail Space
![space](https://user-images.githubusercontent.com/24797493/163417423-c67be47d-aeae-4305-a861-42d4ffcc1a13.png)

# Existing style guide
Wagtail has a contrib module that enables a style guide to be displayed in the admin. But, that appears to be a bit outdated (e.g. the new color of the sidebar and the text does not show up in the color pallet). 

This is definitely a good start. It does need updating, and also might need some extra styles more targeted at prose.

Also, is there a good way to be able to share the CSS already build for Wagtail? 

It might also be nice to make the styleguide available outside of a Wagtail admin.
![styleguide](https://user-images.githubusercontent.com/24797493/163417672-51811a8a-a9e6-49ba-be7c-7fcccd389509.png)

**Comment by thibaudcolas** [comment]
This is all already well under way as part of the page editor project, though we haven’t been documenting or sharing our progress on this front that much.

- Design tokens setup: https://github.com/wagtail/wagtail/issues/7978
- The styleguide is being replaced with Storybook: https://github.com/wagtail/wagtail/pull/8304
- We’ve changed our brand font (#7724) to one that should be simpler to reuse outside of Wagtail

**Reply by tbrlpld** [reply]
Looking forward to having that available!

It might be a nice idea (as @jacobtoppm had suggested) to maybe make the Storybook setup available publicly and link to it from the docs or Wagtail.org

**Comment by tbrlpld** [comment]
Here is the [Figma community version of the Wagtail design system](https://www.figma.com/community/file/1098619062654073532).
